### PR TITLE
update Docker examples to specify HOSTNAME properly

### LINKED ISSUE #58657: AWS ECS Docker deploy not working due to HOSTNAME env variable being overridden
### Verify canary release

- [ ] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Binaries:
  Node: 18.16.0
  npm: 9.5.1
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 13.5.6
  eslint-config-next: 13.0.4
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 4.9.3
Next.js Config:
  output: standalone
```

### Which example does this report relate to?

with-docker

### What browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

### Describe the Bug

#44627 allowed the user to specify a `$HOSTNAME` environment variable. This seems to break deployment to AWS ECS unless you explicitly set `HOSTNAME=0.0.0.0` in the task definition configuration because by default the AWS ECS instance sets $HOSTNAME to what looks to be the container id.

The `with-docker` example has been updated to include:
`ENV HOSTNAME "0.0.0.0"`

This, however, does not seem to work as `0.0.0.0` is the default value and so it can still be overwritten (which it is by AWS) to the container id. The only fix I have found to work is to explicitly set the environment variable in the task.

I'm not sure the best way to fix this, maybe just document the behavior? I am unfamiliar with a method in a docker that lets you set an environment variable such that it _can't be overriden_. One way to fix it would be to change the variable name to `NEXT_HOSTNAME`, but is it too late given that this already shipped?

### Expected Behavior

Running the standalone server app with `node server.js` should start the server. Instead you get the error message described in this comment: https://github.com/vercel/next.js/issues/54093#issuecomment-1684912389

### To Reproduce

Deploy the with-docker container example to ECS with the latest version of next.js

### PR DESCRIPTION
**Pull request by fursich**
### What?

Update docker examples to correctly set HOSTNAME env variable for standalone output

Related to PR:
https://github.com/vercel/next.js/pull/52804

### TL;DR

The dockerfiile examples need to modified as that contains HOSTNAME setting that apparently is not working as expected.

Related to Issue:
https://github.com/vercel/next.js/issues/58657

Affected examples:
https://github.com/vercel/next.js/tree/canary/examples/with-docker
https://github.com/vercel/next.js/tree/canary/examples/with-docker-multi-env

### Longer explanations

Based on [Docker document](https://docs.docker.com/engine/reference/builder/#environment-replacement):
> You can also use environment variables with RUN, CMD, and ENTRYPOINT instructions, but in those cases the variable substitution is handled by the command shell, not the builder.

It means that when executing the last CMD ( `node server.js` ) in Dockerfile samples, the HOSTNAME defined by prior `ENV` instruction is simply ignored.

This causes problems typically when the host process sets HOSTNAME - e.g. for instance when using AWS Fargate for deployment [we can't have control over HOSTNAME](https://stackoverflow.com/a/52871552) set by the host process. (also refer to the issue #58657 above)

### How?

Updated Dockerfilie samples, by setting HOSTNAME directly in CMD instruction that launches nextjs server, not in the builder process.

Used this setting at my end (with AWS Fargate) to confirm that it fixes the network problem.

Closes NEXT-
Fixes #58657

### CODE REVIEW

File: examples/with-docker-multi-env/docker/development/Dockerfile [hunk 1]
```diff
@@ -47,6 +47,5 @@ USER nextjs
 EXPOSE 3000
 
 ENV PORT 3000
-ENV HOSTNAME localhost
 
-CMD ["node", "server.js"]
+CMD HOSTNAME=localhostt node server.js
```
Comment (tanishi) [inline examples/with-docker-multi-env/docker/development/Dockerfile:51]: Excuse me for interrupting.
Is there a typo?

Comment (fursich) [reply examples/with-docker-multi-env/docker/development/Dockerfile]: Oops my bad..thank you for the catch! Updated!

**Comment by akhlivniukallane** [comment]
To give an example of how harmful that is: we couldn't find out why we can't use any Next.js version above 13.4.12 on a self-hosted solution. This change with HOSTNAME appeared on 13.4.13 without mentioning that in the release changes. Example silently updated. It was updated not fully correctly and didn't work correctly in a long run. This PR has been open for three months with working solution, and the only way to find the reason for failing container startups is to google this issue with that solution. Self-hosting Next.js is tough :(

**Review by samcx — APPROVED** [review]
Thank you for submitting a <picture data-single-emoji=":pr:" title=":pr:">:pr:</picture>!

<picture data-single-emoji=":lgtm:" title=":lgtm:">:lgtm:</picture>
